state/model.go

```diff
@@ -356,10 +356,18 @@ func (st *State) NewModel(args ModelArgs) (_ *Model, _ *State, err error) {
 	// specified, that the cloud supports the "empty" authentication
 	// type.
 	owner := args.Owner
-	cloudCredentials, err := st.CloudCredentials(owner, args.CloudName)
+	storedCredentials, err := st.CloudCredentials(owner, args.CloudName)
 	if err != nil {
 		return nil, nil, errors.Trace(err)
 	}
+
+	cloudCredentials := make(map[string]jujucloud.Credential, len(storedCredentials))
```

Comment:
We shouldn't be doing this here as we're still in state. We pass cloudCredentials to validateCloudCredentials() and that's a state method so should take state.CloudCredential values

Reply:
I agree that we should not. However, for historic reasons and because we use it from Iniitialize this method works on cloud.Credentials. 
I am happy to fix it as a separate PR since this one is getting too big and is loosing focus.

Reply:
Can we add a TODO to the validate method?